helm/cluster/values.schema.json

```
@@ -1399,6 +1399,12 @@
                                             ],
                                             "maxItems": 1,
                                             "minItems": 1
+                                        },
+                                        "nodeCidrMaskSize": {
+                                            "type": "integer",
```

We could ensure it's in some reasonable range. I invented those values, feel free to change to 1 and 32 🤷:

```suggestion
                                            "type": "integer",
                                            "minimum": 16,
                                            "maximum": 27,
```

---

Let's also maybe add a CI `ci/test-*-values.yaml` file which overrides these values, just to check this feature keeps working

---

I guess the theoretical usable number is /27 with 30 usable addresses, as we have around 17 pods in `kube-system` on control-plane nodes. It wouldn't be particularly _useful_, but definitely _usable_

---

In the other direction though, the maximum number of pods per node is 256. This is according to docs on a [deprecated flag though](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/). The [KubeletConfiguration](https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/) parameters no longer specify a maximum. So considering that I think /16 is big enough as a maximum here :shrug: